src/App.tsx

```diff
@@ -1,156 +1,74 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState } from 'react';
+
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList/TodoList';
+import { Filters, TodoFilter } from './components/TodoFilter/TodoFilter';
+import { TodoForm } from './components/TodoForm/TodoForm';
+import { TodoError } from './components/TodoError/TodoError';
+
+import { useTodos } from './hooks/useTodos';
 
-const USER_ID = 0;
+const USER_ID = 10538;
 
 export const App: React.FC = () => {
+  const [activeFilter, setActiveFilter] = useState<Filters>('all');
+
+  const { todos, error } = useTodos(USER_ID);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const activeTodos = todos.filter(todo => !todo.completed);
```

Comment:
I would personally do something like this:
![image](https://github.com/mate-academy/react_todo-app-loading-todos/assets/87986955/7595d84f-4e4f-4a17-ae87-05eca3ca2fa3)
![image](https://github.com/mate-academy/react_todo-app-loading-todos/assets/87986955/3b4844cb-c814-4262-b947-6ec3c6eac3e0)

Currently you are doing the filtering everytime the `activeFilter` changes, with `useMemo` it will only recalculate when the todos fetched from the server changes, also with this implementation you are using only one loop instead iterating over todos list two times. If you were writing an application that fetches hundreds of data it would be very beneficial.

Reply:
Ohh I see. That's nice